### ISSUE TITLE
UPE - Checkout with cart total under 50 cents

### ISSUE THREAD

**Issue by bborman22** [woocommerce/woocommerce-gateway-stripe#1769]
**Describe the bug**
This is a reference to the [WC Pay issue](https://github.com/Automattic/woocommerce-payments/issues/2700) and any solutions or thoughts on it could be valuable here as well. When trying to checkout with a cart total under 50 cents USD, UPE throws an error when trying to mount. When this was experienced by merchants using WC Pay with UPE they called out that this 50 cent limit should only be applied after shipping cost has been added in. Due to how UPE works though, we need to create the intent before we know the shipping costs which will cause this error. We will need to find a workaround for this issue so that it does not throw an error until shipping costs are calculated in to the total.

**To Reproduce**
- Start with a store with this branch and UPE enabled (_wcstripe_feature_upe set to 'yes' in wp_options table) and the new checkout experience enabled from the WC Stripe settings page.
- Create a product that is under 50 cents USD.
- As customer, add this product to cart and go to checkout.
- Confirm that the error message: Amount must be at least $0.50 usd is displayed on the checkout page. (this message will only display after #1768 is merged, until then a more generic error will display).

**Expected behavior**
Stripe does not handle transactions under 50 cents. A few merchants mentioned that this error should only display if the total is under 50 cents once shipping costs have been factored in. So the expected behavior is to allow the customer to continue with checkout until a shipping option has been selected and only then display the 50 cent limit error.

**Screenshots**
![UPE-50cent-error](https://user-images.githubusercontent.com/68524302/131568016-976de16d-30ce-47d1-8e8d-740567c46187.png)

**Comment by reykjalin** [comment]
Just as an idea on how to potentially solve this:

Could we send a request for the cart and only create the payment intent after we receive that response? We have an AJAX endpoint set up that I believe should be able to do this properly 🤔

**Comment by leonardola** [comment]
This also happens with [Boleto and OXXO](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2115#discussion_r750701224)

**Comment by mattallan** [comment]
I've confirmed our deferred intent changes have fixed this issue. Closing this!

Flows I ran through to confirm it's safe to close this one:
- With the new checkout experience enabled
  - 👍 $0.40 with no shipping (error notice on checkout showing minimum $0.50)
  - 👍 $0.40 with $1 shipping
  - 👍 EUR APMs (SEPA, ideal)
  - 👍 Voucher APMs (OXXO, boleto)
  - 👍 Apple pay/google pay
  - 👍 Saved cards and new cards
  - 👍 tested on shortcode checkout and block checkout
- With legacy checkout experience
  - 👍 $0.40 with no shipping (error notice on checkout showing minimum $0.50)
  - 👍 $0.40 with $1 shipping
